stempy/main.cpp

```diff
@@ -12,24 +64,98 @@ int main (int argc, char *argv[])
     return 1;
   }
 
-  stempy::StreamReader reader(argv[1]);
+  string inputPath = argv[1];
+
+  // check the input path
+  if (checkIfDirectory(inputPath))
+  {
+    cout << endl << "Input path " << inputPath << " is valid, starting processing" << endl;
+    // get all the files in this path
+    const vector<string> files = GetDirectoryFiles(inputPath);
+    int numFiles = files.size();
+    cout << numFiles << " files have been found" << endl;
 
-  int count = 0;
-  while (true) {
-    auto stream = reader.read();
+    // all the STEM images that would be summed into one final STEM image
+    vector<STEMImage> allPartialSTEMImages;
+
+    // information about processing
+    int width = 160;
+    int height = 160;
+    // -1 indicate the center of the image
+    int centerX = -1;
+    int centerY = -1;
+
+    for (int i = 0; i < numFiles; i++)
+    {
+      cout << "Processing " << i+1 << "/" << numFiles << ": " << files[i] << endl;
+      string curFile = files[i];
+      // read the data
+      stempy::StreamReader reader(inputPath+curFile);
+      auto stream = reader.read();
```

Comment:
Calling `read()` on the `StreamReader` reads one block and increments the underlying `ifstream`. You might need to call `reader.reset()` to reset it to the beginning if you are going to use it again later.

Reply:
I might be wrong but I think that for each iteration, reader is newly created and gets a new file, instead of getting a single block. In this case, do we still need to reset it?

Reply:
Yes, that is true. But I was referring more to [this line](https://github.com/OpenChemistry/stempy/pull/59/files#diff-f87a2c458e9afaf9bbf1d223829b9e4dR130) below where the reader gets re-used again. You might need to call `reader.reset()` before that line, because the `StreamReader::begin()` function doesn't actually reset the `ifstream` position, but starts reading from wherever it currently is.

It might be more intuitive for us to get `StreamReader::begin()` to start from the beginning, though. What do you think, @cjh1?

Reply:
(Note that `StreamReader::begin()` is [here](https://github.com/OpenChemistry/stempy/blob/master/stempy/reader.h#L55))